test/e2e/upgrade/upgrade_test.go

```diff
@@ -0,0 +1,55 @@
+package upgrade
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	nropv1 "github.com/openshift-kni/numaresources-operator/api/numaresourcesoperator/v1"
+	"github.com/openshift-kni/numaresources-operator/internal/api/annotations"
+	nropmcp "github.com/openshift-kni/numaresources-operator/internal/machineconfigpools"
+	"github.com/openshift-kni/numaresources-operator/pkg/objectnames"
+	e2eclient "github.com/openshift-kni/numaresources-operator/test/utils/clients"
+	"github.com/openshift-kni/numaresources-operator/test/utils/objects"
+	machineconfigv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+)
+
+var _ = Describe("[Upgrade]", func() {
+	var initialized bool
+
+	BeforeEach(func() {
+		if !initialized {
+			Expect(e2eclient.ClientsEnabled).To(BeTrue(), "failed to create runtime-controller client")
+		}
+		initialized = true
+	})
+
+	Context("after operator upgrade", func() {
```

Comment:
something is missing, maybe here maybe in general. What documents or better ensures this lane is running against an upgrade?

Reply:
This test suite was born from a use-case which is testing that 4.1X to 4.18 upgrade results with the removal of the NROP MC.
It is expected to be invoked by this new lane:
https://github.com/openshift-kni/numaresources-operator/pull/1073

Reply:
which is fine, can we encode this as prerequisite for the test cases?
e.g. this test should skip (or fail) if running on a cluster on which NROP 4.17 or earlier is deployed

Reply:
We can do that. 
I was thinking that since we merge this to the main branch and probably backport only to 4.18 branch, we would expect this to be running only agains clusters with OCP 4.18 and above.
Therefore, we don't need to check this as a prerequisite.

Does my assumptions are incorrect?
Anyhow let me know if you'd still prefer to have this check

Reply:
I'd still be happier if we have something automated which enforces/reminds the prerequisites and the purpose of the tests. Doesn't need something complex, but should be something automated so we don't _just_ rely on our memory.